android/runtime/v8/src/native/TypeConverter.cpp

```diff
@@ -163,6 +163,30 @@ Local<Value> TypeConverter::javaStringToJsString(Isolate* isolate, JNIEnv *env,
 	return jsString;
 }
 
+Local<Value> TypeConverter::javaBytesToJsString(Isolate* isolate, jbyteArray javaBytes)
+{
+	JNIEnv *env = JNIScope::getEnv();
+	if (env == NULL) {
+		return String::Empty(isolate);
+	}
+	return TypeConverter::javaBytesToJsString(isolate, env, javaBytes);
+}
+
+Local<Value> TypeConverter::javaBytesToJsString(Isolate* isolate, JNIEnv *env, jbyteArray javaBytes)
+{
+	if (!javaBytes) {
+		return Null(isolate);
+	}
+
+	jboolean isCopy;
+	jsize nativeBytesLength = env->GetArrayLength(javaBytes);
+	uint8_t* nativeBytes = reinterpret_cast<uint8_t*>(env->GetByteArrayElements(javaBytes, &isCopy));
+	Local<String> jsString = v8::String::NewFromOneByte(V8Runtime::v8_isolate, nativeBytes, v8::NewStringType::kNormal, nativeBytesLength).ToLocalChecked();
```

Comment:
Ok, so having just done the node string_encoder/Buffer shims this raises a red flag for me. Basically you're making an implicit assumption that the byte array contains a string with a one-byte charset (i.e. ascii, latin-1, binary). That seems like a big no-no here. I think any UTF-8 content outside the typical ASCII range would break this, no? I just think you probably have been lucky and only tried the overlapping character range that exists in both ASCII and UTF-8 and requires only 1 byte to "encode" in UTF-8 - like here: https://ascii.cl
Write some JS code with some umlauts/euros/stuff in strings or identifiers and I'll bet you end up with some garbled source out at the end here.

I would think you'd need to pass along the original encoding of the file plus the bytes that make it up. In reality, I would think UTF-8 would be a good assumption for a charset, but if you had any input files that were say utf-16 you'd have to transcode before encrypting, etc. (And you're already making some assumptions that the bytes are actually utf-8 in some Java code I see in this PR). Note that UTF-8 is the most widely used and for good reason as it packs the most common characters into 1 byte and can grow up to 4 bytes for the esoteric characters (whereas UTF-16 uses 2 or 4 bytes, so is less efficient for common ASCII chars in source code)
(Also, for reference Java uses UTF-16 for strings normally, and so does Javascript - so passing an actual byte array through here as you do should be more efficient memory-wise)

So I guess my suggestion here would be:
- have your encryption generate a byte array, sure, but make sure it holds the utf-8 encoded character data
- change this to ` v8::String::NewFromUtf8()`

Reply:
So I done a quick test using this test case:
```JS
const win = Ti.UI.createWindow({ backgroundColor: 'white' });
const lbl = Ti.UI.createLabel({
    color: 'black',
    font: { fontSize: '18pt' },
    text: 'Let\'s test UTF-8 characters: \nCopyright: ©\nBritish Pount: £\nEuro: €\nGreek: αβγπ\nTamil: ௵௶௷\nEmoji: ✌😁😺🍔📦'
});

// Let's also test UTF-8 characters in a comment: అೠቜ

win.add(lbl);
win.open();
```

And it works correctly, this is because the byte array is already guaranteed to be UTF-8 encoded before being passed in. But ideally I should be using `v8::String::NewFormUtf8` as `v8::String::NewFromOneByte` assumes Latin-1 data.